pkgsMusl.xterm: build fails

### Steps To Reproduce
1. nix build .#pkgsMusl.xterm

### Build log
```
./main.c:805:34: error: ‘TAB3’ undeclared here (not in a function); did you mean ‘TAB0’
```
Full logs: https://termbin.com/0znw

### Additional context
Errors happen when building for musl, dynamic.

### Notify maintainers
@nequissimus @vrthra

--- a/pkgs/applications/terminal-emulators/xterm/default.nix
+++ b/pkgs/applications/terminal-emulators/xterm/default.nix
@@ -54,8 +54,13 @@ stdenv.mkDerivation rec {
     "--with-app-defaults=$(out)/lib/X11/app-defaults"
   ] ++ lib.optional enableDecLocator "--enable-dec-locator";
 
-  # Work around broken "plink.sh".
-  NIX_LDFLAGS = "-lXmu -lXt -lICE -lX11 -lfontconfig";
+  env = {
+    # Work around broken "plink.sh".
+    NIX_LDFLAGS = "-lXmu -lXt -lICE -lX11 -lfontconfig";
+  } // lib.optionalAttrs stdenv.hostPlatform.isMusl {
+    # Various symbols missing without this define: TAB3, NLDLY, CRDLY, BSDLY, FFDLY, CBAUD
+    NIX_CFLAGS_COMPILE = "-D_GNU_SOURCE";
+  };
 
   # Hack to get xterm built with the feature of releasing a possible setgid of 'utmp',
   # decided by the sysadmin to allow the xterm reporting to /var/run/utmp

